src/dune_engine/action_intf.ml

```diff
@@ -39,7 +39,7 @@ module type Ast = sig
     | Copy_and_add_line_directive of path * target
     | System of string
     | Bash of string
-    | Write_file of target * string
+    | Write_file of target * string * bool
```

Comment:
Would be better if this argument was a record. It's hard to guess what the naked bool stands for otherwise.

Reply:
Do you mean
```suggestion
    | Write_file of target * string * {binary: bool}
```
or
```suggestion
    | Write_file of { target: target
                    ; content: string
                    ; binary: bool
                    }
```

Reply:
I meant the latter. `content` is a useful clarification as well.

Reply:
agreed